internal/mode/static/state/graph/httproute_test.go

```diff
@@ -517,7 +567,76 @@ func TestBuildRoute(t *testing.T) {
 					},
 				},
 			},
-			name: "invalid with invalid and valid rules",
+			name: "invalid filter with invalid and valid rules",
+		},
+		{
+			validator: validatorInvalidFieldsInRule,
+			hr:        hrInvalidValidFilter,
+			expected: &Route{
+				Source: hrInvalidValidFilter,
+				Valid:  true,
+				ParentRefs: []ParentRef{
+					{
+						Idx:     0,
+						Gateway: gatewayNsName,
+					},
+				},
+				Conditions: []conditions.Condition{
+					staticConds.NewRoutePartiallyInvalid(
+						`Dropped Rule(s): ` +
+							`spec.rules[1].filters[0].requestRedirect.hostname: Invalid value: ` +
+							`"invalid.example.com": invalid hostname`,
+					),
+				},
+				Rules: []Rule{
+					{
+						ValidMatches: true,
+						ValidFilters: true,
+					},
+					{
+						ValidMatches: true,
+						ValidFilters: false,
+					},
+				},
+			},
+			name: "invalid and valid filter with valid rules",
+		},
+		{
+			validator: validatorInvalidFieldsInRule,
+			hr:        hrInvalidValidRuleAndFilter,
+			expected: &Route{
+				Source: hrInvalidValidRuleAndFilter,
+				Valid:  true,
+				ParentRefs: []ParentRef{
+					{
+						Idx:     0,
+						Gateway: gatewayNsName,
+					},
+				},
+				Conditions: []conditions.Condition{
+					staticConds.NewRoutePartiallyInvalid(
+						`Dropped Rule(s): ` +
+							`[spec.rules[0].matches[0].path.value: Invalid value: "/invalid": invalid path, ` +
+							`spec.rules[2].filters[0].requestRedirect.hostname: Invalid value: ` +
+							`"invalid.example.com": invalid hostname]`,
+					),
+				},
+				Rules: []Rule{
+					{
+						ValidMatches: false,
+						ValidFilters: true,
+					},
+					{
+						ValidMatches: true,
+						ValidFilters: true,
+					},
+					{
+						ValidMatches: true,
+						ValidFilters: false,
+					},
+				},
+			},
+			name: "invalid filter and rules with valid filter and rules",
```

Comment:
this repeats the case above
`name: "invalid filter with invalid and valid rules",`

Reply:
Removed this test case as it does seem to be testing the same case with a minor difference.